ethclient/ethclient.go

```diff
@@ -506,6 +507,35 @@ func (ec *Client) SendTransaction(ctx context.Context, tx *types.Transaction) er
 	return ec.c.CallContext(ctx, nil, "eth_sendRawTransaction", common.ToHex(data))
 }
 
+// PreparePrivateTransaction sends the encoded raw transaction to transaction manager,
+// returning the hash of private transaction.
+func (ec *Client) PreparePrivateTransaction(ctx context.Context, encodedTx hexutil.Bytes, privateFrom string, privateFor []string) (hexutil.Bytes, error) {
+	if privateFor == nil {
+		return nil, errors.New("privateFor cannot be nil")
+	}
+	tx := new(types.Transaction)
+	if err := rlp.DecodeBytes(encodedTx, tx); err != nil {
+		return nil, err
+	}
+
+	if private.P == nil {
+		return nil, errors.New("transaction manager not set up")
+	}
+	data, err := private.P.Send(tx.Data(), privateFrom, privateFor)
+	if err != nil {
+		return nil, err
+	}
+
+	tx.SetPrivate()
+	tx.SetData(data)
```

Comment:
Use `types.NewTransaction` or `types.NewCreateTransaction` instead

Reply:
I have removed SetData and use types.NewTransaction/types.NewContractCreation to create private transaction